arangod/Aql/TraversalExecutor.cpp

```diff
@@ -144,81 +198,451 @@ TraversalExecutorInfos::filterConditionVariables() const {
   return _filterConditionVariables;
 }
 
+TraverserOptions::UniquenessLevel TraversalExecutorInfos::getUniqueVertices()
+    const {
+  return _uniqueVertices;
+}
+
+TraverserOptions::UniquenessLevel TraversalExecutorInfos::getUniqueEdges()
+    const {
+  return _uniqueEdges;
+}
+
+TraverserOptions::Order TraversalExecutorInfos::getOrder() const {
+  return _order;
+}
+
+void TraversalExecutorInfos::setOrder(TraverserOptions::Order order) {
+  _order = order;
+}
+
+bool TraversalExecutorInfos::isRefactor() const { return _refactor; }
+
+transaction::Methods* TraversalExecutorInfos::getTrx() { return _trx; }
+arangodb::aql::QueryWarnings& TraversalExecutorInfos::getWarnings() {
+  return _query.warnings();
+}
+
+std::pair<arangodb::graph::VertexUniquenessLevel,
+          arangodb::graph::EdgeUniquenessLevel>
+TraversalExecutorInfos::convertUniquenessLevels() const {
+  // TODO [GraphRefactor]: This should be a temoporary function as we remove
+  // TraverserOptions in total after the graph refactor is done.
+  auto vertexUniquenessLevel = graph::VertexUniquenessLevel::NONE;
+  auto edgeUniquenessLevel = graph::EdgeUniquenessLevel::NONE;
+
+  if (getUniqueVertices() == traverser::TraverserOptions::PATH) {
+    vertexUniquenessLevel = graph::VertexUniquenessLevel::PATH;
+    edgeUniquenessLevel = graph::EdgeUniquenessLevel::PATH;
+  } else if (getUniqueVertices() == traverser::TraverserOptions::GLOBAL) {
+    vertexUniquenessLevel = graph::VertexUniquenessLevel::GLOBAL;
+    edgeUniquenessLevel = graph::EdgeUniquenessLevel::PATH;
+  }
+
+  if (getUniqueEdges() == traverser::TraverserOptions::PATH) {
+    edgeUniquenessLevel = graph::EdgeUniquenessLevel::PATH;
+  } else if (getUniqueEdges() == traverser::TraverserOptions::GLOBAL) {
+    edgeUniquenessLevel = graph::EdgeUniquenessLevel::PATH;
+  }
+
+  return std::make_pair(vertexUniquenessLevel, edgeUniquenessLevel);
+}
+
+// TODO [GraphRefactor]: Add a parameter to toggle tracing variants of
+// enumerators.
+auto TraversalExecutorInfos::parseTraversalEnumerator(
+    TraverserOptions::Order order,
+    TraverserOptions::UniquenessLevel uniqueVertices,
+    TraverserOptions::UniquenessLevel uniqueEdges, double defaultWeight,
+    std::string const& weightAttribute, arangodb::aql::QueryContext& query,
+    arangodb::graph::BaseProviderOptions&& baseProviderOptions,
+    arangodb::graph::PathValidatorOptions&& pathValidatorOptions,
+    arangodb::graph::OneSidedEnumeratorOptions&& enumeratorOptions) -> void {
+  // TODO [GraphRefactor]: Temporary try to minimize copy-paste-tank, but
+  // failed. auto [vertexUnique, edgeUnique] = convertUniquenessLevels();
+
+  if (order == TraverserOptions::Order::DFS) {
+    switch (uniqueVertices) {
+      case TraverserOptions::UniquenessLevel::NONE:
+        switch (uniqueEdges) {
+          case TraverserOptions::NONE:
```

Comment:
```suggestion
          case TraverserOptions::UniquenessLevel::NONE:
```

please fix all cases! All of them use the wrong enum I think

Reply:
Done thanks.